core/src/main/scala/org/apache/pekko/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -50,49 +54,33 @@ class JournalQueries(
     }
   }
 
-  private def selectAllJournalForPersistenceIdDesc(persistenceId: Rep[String]) =
-    selectAllJournalForPersistenceId(persistenceId).sortBy(_.sequenceNumber.desc)
-
   private def selectAllJournalForPersistenceId(persistenceId: Rep[String]) =
     JournalTable.filter(_.persistenceId === persistenceId).sortBy(_.sequenceNumber.desc)
 
   def delete(persistenceId: String, toSequenceNr: Long) = {
     JournalTable.filter(_.persistenceId === persistenceId).filter(_.sequenceNumber <= toSequenceNr).delete
   }
 
-  def markJournalMessagesAsDeleted(persistenceId: String, maxSequenceNr: Long) =
+  def markSeqNrJournalMessagesAsDeleted(persistenceId: String, sequenceNr: Long) =
     JournalTable
       .filter(_.persistenceId === persistenceId)
-      .filter(_.sequenceNumber <= maxSequenceNr)
+      .filter(_.sequenceNumber === sequenceNr)
```

Comment:
why this is changed to  `===`? i think this should not be safe if there are concurrent running

Reply:
I can't understand you, this is the ORM method. Except for the highest sequence number that needs to be retained for querying from pid, all other rows will be deleted, so why do we need to update them?